cmd/cluster-kube-apiserver-operator/render/render.go

```diff
@@ -62,176 +53,98 @@ func NewRenderCommand() *cobra.Command {
-		Image:                 r.manifest.image,
-		ImagePullPolicy:       r.manifest.imagePullPolicy,
-		ConfigHostPath:        r.manifest.configHostPath,
-		ConfigFileName:        r.manifest.configFileName,
-		CloudProviderHostPath: r.manifest.cloudProviderHostPath,
-		SecretsHostPath:       r.manifest.secretsHostPath,
-		LockHostPath:          r.manifest.lockHostPath,
-		EtcdServerURLs:        r.manifest.etcdServerURLs,
-		EtcdServingCA:         r.manifest.etcdServingCA,
+	if err := r.generic.Complete(); err != nil {
+		return err
 	}
+	return nil
+}
 
-	// create post-poststrap configuration
-	var err error
-	renderConfig.PostBootstrapKubeAPIServerConfig, err = r.configFromDefaultsPlusOverride(&renderConfig, filepath.Join(r.templatesDir, "config", "config-overrides.yaml"))
+type TemplateData struct {
+	genericrenderoptions.ManifestConfig
+	genericrenderoptions.FileConfig
 
-	// load and render templates
-	if renderConfig.Assets, err = assets.LoadFilesRecursively(r.assetInputDir); err != nil {
-		return fmt.Errorf("failed loading assets from %q: %v", r.assetInputDir, err)
-	}
-	for _, manifestDir := range []string{"bootstrap-manifests", "manifests"} {
-		manifests, err := assets.New(filepath.Join(r.templatesDir, manifestDir), renderConfig, assets.OnlyYaml)
-		if err != nil {
-			return fmt.Errorf("failed rendering assets: %v", err)
-		}
-		if err := manifests.WriteFiles(filepath.Join(r.assetOutputDir, manifestDir)); err != nil {
-			return fmt.Errorf("failed writing assets to %q: %v", filepath.Join(r.assetOutputDir, manifestDir), err)
-		}
-	}
+	// LockHostPath holds the api server lock file for bootstrap
+	LockHostPath string
 
-	// create bootstrap configuration
-	mergedConfig, err := r.configFromDefaultsPlusOverride(&renderConfig, filepath.Join(r.templatesDir, "config", "bootstrap-config-overrides.yaml"))
-	if err != nil {
-		return fmt.Errorf("failed to generated bootstrap config: %v", err)
-	}
-	if err := ioutil.WriteFile(r.configOutputFile, mergedConfig, 0644); err != nil {
-		return fmt.Errorf("failed to write merged config to %q: %v", r.configOutputFile, err)
-	}
+	// EtcdServerURLs is a list of etcd server URLs.
+	EtcdServerURLs []string
 
-	return nil
+	// EtcdServingCA is the serving CA used by the etcd servers.
+	EtcdServingCA string
 }
 
-func (r *renderOpts) configFromDefaultsPlusOverride(data *Config, tlsOverride string) ([]byte, error) {
-	defaultConfig := v311_00_assets.MustAsset(filepath.Join(bootstrapVersion, "kube-apiserver", "defaultconfig.yaml"))
-	bootstrapOverrides, err := readFileTemplate(tlsOverride, data)
-	if err != nil {
-		return nil, fmt.Errorf("failed to load config override file %q: %v", tlsOverride, err)
-	}
-	// TODO: Remove this when the flag is gone
-	if len(r.deprecatedConfigOverrideFile) > 0 {
-		r.configOverrideFiles = append(r.configOverrideFiles, r.deprecatedConfigOverrideFile)
+// Run contains the logic of the render command.
+func (r *renderOpts) Run() error {
+	if err := r.Complete(); err != nil {
```

Comment:
That's unusual and definitely not in-line with all of our CLI. We usually call the Validate-Complete-Run triad inside of cobra's Run, I think that should be true here as well.

Reply:
done